VAR-310 | Add calendar into project

Jokinen opened the pull request:
The goal of this PR is to add the calendar component from Varaamo with as few steps as possible and replace the current "left side" of the application with this calendar.

You can find the visuals from here: https://projects.invisionapp.com/share/DMVSOEHYTUP#/screens

The meat of this PR is the addition of the Calendar component from Varaamo into this repo. Here are the general steps I took to integrate the component with this project:
1) Copied over component file
2) Copied over utilities
3) Removed unused utilities
4) Removed tests. For some reason I couldn't get the tests to run. They can be added later on.
5) Transformed sass files into styled-component components
6) Integrated translations
7) Created interop between Calendar component and the Immutable.js data structure this app uses

Jokinen committed:
Use prop-types from separate package

Jokinen committed:
Remove eslint rule that conflicts with prettier

Conflict here means that what was correct formattting for one library,
was incorrect for the other and vice versa.

Jokinen committed:
Add initial day calendar view

In these changes we are importing the Fullcalendar component we use in
Varaamo. I attempted to make the addition with minimal required
changes. Some polishing is still needed, but I should be able to make
those when I am implementing the week view.

I also shuffled the version we are using of some libraries, because
Fullcalendar was dependent on these.

Jokinen committed:
Resolve lint issues in Booking component

tuovinensanttu commented on app/components/AreaBooking/index.js:
```diff
@@ -17,112 +18,132 @@ import reducer from 'containers/HomePage/reducer';
+  resourceId,
+}) => {
+  const wrapperRef = useRef(null);
+  const [, height] = useElementSize(wrapperRef);
+
```
Is the  `,` in front of height needed or just a typo?

Jokinen replied:
It's syntax for destructuring an array without declaring a variable for array members I don't have a use for.

So it's the same as `[width, height]`, but with width ignored. An unused variable would cause a linting error. The purpose is similar to how `_` is used for instance in Haskell or some other fp oriented languages.

Here's a link to the MDN docs for more details: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring
"Ignoring some returned values"

tuovinensanttu commented on app/components/Calendar/index.js:
```diff
@@ -0,0 +1,442 @@
+      // mobile view config
+      view = 'timeGridDay';
+      headerConfig = {
+        left: 'myPrev,myNext,myToday',
```
These buttons are commented out on initial state. If it's intended to never show these, I suggest you comment these out as well. Even though tablet in portrait mode will probably have more width than 768, you never know what users manage to achieve. :smile:

tuovinensanttu commented on app/components/Calendar/index.js:
```diff
@@ -0,0 +1,442 @@
+import CalendarLegend from './legend/CalendarLegend';
+import {
+  isTimeRangeReservable,
+  getReservationPrice,
```
`getReservationPrice` doesn't exist.

Jokinen replied:
Good catch. I removed the support for price, because it won't likely have many uses in this context.

Really odd that my IDE's linter did not catch this--nor did the linting scrip that the project runs on every commit.

tuovinensanttu commented on app/components/Calendar/messages.js:
```diff
@@ -0,0 +1,28 @@
+  selectedDateLabel: {
+    id: 'Calendar.selectedDateLabel',
+    defaultMessage: 'Valittu aika:',
+  },
```
I might be a little too picky, but would it be a good idea to keep defaultMessage language consistent?

Jokinen replied:
Good point. I changed them to Finnish because that's used elsewhere in the app.

tuovinensanttu reviewed (approved):
Left few comments, but otherwise everything looks good :+1:

Jokinen committed:
Hide left buttons for bigger screens

Jokinen committed:
Remove price support from calendar

Resource price check requires a query to the API. As this functionality
is not specced and it likely won't be supported, because it'd be
difficult to facilitate the passing of payment informaiton through the
tablet, I'm just removing this functionality.

Jokinen committed:
Make default message language consistent

Finnish is used in other parts of the app.

Jokinen commented:
I would have liked to merge https://github.com/City-of-Helsinki/reservation-screen-ui/pull/14 before merging this PR in order to get assurances that this PR passes the CI. But I need these changes to start on the task that's prioritised to be started next, so I am just merging it now.
